crates/sui-types/src/multisig.rs

```diff
@@ -360,11 +360,16 @@ impl MultiSigPublicKey {
                 .map(|w| *w as ThresholdUnit)
                 .sum::<ThresholdUnit>()
                 < threshold
+            || pks
+                .iter()
+                .enumerate()
+                .any(|(i, pk)| pks.iter().skip(i + 1).any(|other_pk| *pk == *other_pk))
         {
             return Err(SuiError::InvalidSignature {
                 error: "Invalid multisig public key construction".to_string(),
             });
         }
+
         Ok(MultiSigPublicKey {
             pk_map: pks.into_iter().zip(weights.into_iter()).collect(),
             threshold,
```

Comment:
nit: same for construct() - should it just invoke new() and call validate?

Reply:
We can't call validate() here without refactoring it, because the MultiSigPublicKey object is not yet created at the time we call new(), but validate() needs it to be created. Since we want to have minimum changes - we decided to extend checks.

> instead checking above for validity, we can just call .validate here

Reply:
> nit: same for construct() - should it just invoke new() and call validate?

Same for the construct() function. Also, the construct() function is one of the parts of multisig_legacy implementation and such changes potentially may cause a problems with it. Same as for refactoring - since we wanted to keep minimum changes.